### PR TITLE
`use_self` for tuple structs

### LINKED ISSUE #3498: use_self does not diagnose tuple structs
[Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=fdfb70343ce5f8be63662ad6467aee62):

```rust
pub struct A(i32);

impl A {
    pub fn a() -> Self {
        A(0)  // should print warning: use Self(0)
    }
}
```

does not raise any warnings when the `use_self` lint is enabled.

### PR DESCRIPTION
**Pull request by detrumi**
Fixes #3498

### CODE REVIEW

File: clippy_lints/src/use_self.rs [hunk 1]
```diff
@@ -25,7 +26,12 @@ use if_chain::if_chain;
 /// name
 /// feels inconsistent.
 ///
-/// **Known problems:** None.
+/// **Known problems:**
+/// - Does not trigger within locally defined macros (#2098)
```
Comment (oli-obk) [inline clippy_lints/src/use_self.rs]: I don't think we should list false negatives here.

File: tests/ui/use_self.rs [hunk 2]
```diff
@@ -69,7 +67,8 @@ mod lifetimes {
             Foo { foo_str: "foo" }
         }
 
-        // `Self` is applicable here
+        // TODO: the lint does not handle lifetimed struct
```
Comment (oli-obk) [inline tests/ui/use_self.rs]:
```suggestion
        // FIXME: the lint does not handle lifetimed struct
```

File: clippy_lints/src/use_self.rs (hunk 1 above)
Comment (detrumi) [reply clippy_lints/src/use_self.rs]: Ah, makes sense.

**Pull request merged**